aea/registries/base.py

```diff
@@ -46,43 +47,50 @@
 PACKAGE_NAME_REGEX = re.compile(
     "^([A-Z0-9]|[A-Z0-9][A-Z0-9._-]*[A-Z0-9])$", re.IGNORECASE
 )
-INTERNAL_PROTOCOL_ID = "internal"
+INTERNAL_PROTOCOL_ID = PublicId.from_string("fetchai/internal:0.1.0")
 DECISION_MAKER = "decision_maker"
 
 
-class Registry(ABC):
+Item = TypeVar("Item")
+ItemId = TypeVar("ItemId")
+ComponentId = Tuple[SkillId, str]
+SkillComponentType = TypeVar("SkillComponentType", Handler, Behaviour, Task)
+
+
+class Registry(Generic[ItemId, Item], ABC):
     """This class implements an abstract registry."""
 
     @abstractmethod
-    def register(self, id: Tuple[Any, Any], item: Any) -> None:
+    def register(self, item_id: ItemId, item: Item) -> None:
         """
         Register an item.
 
-        :param id: the identifier of the item.
+        :param item_id: the public id of the item.
         :param item: the item.
         :return: None
+        :raises: ValueError if an item is already registered with that item id.
```

Comment:
the interface says that the implementer should raise a `ValueError` if an item is already registered with that item id.

Notice the asymmetry with `unregister` (that is, we do not care about presence or not presence of the item). That could change though.

Reply:
we should maybe create that symmetry now and add the ValueError to unregister?

Reply:
ok agreed